precompile/contracts/feemanager/contract_test.go

```diff
@@ -165,86 +161,55 @@ func TestFeeManagerRun(t *testing.T) {
 
 				return input
-			}(t),
+			},
 			SuppliedGas: SetFeeConfigGasCost,
 			ReadOnly:    true,
 			ExpectedErr: vmerrs.ErrWriteProtection.Error(),
 		},
 		"readOnly setFeeConfig with allow role fails": {
-			Caller: enabledAddr,
-			Input: func(tt *testing.T) []byte {
+			Caller: allowlist.EnabledAddr,
+			InputFn: func(tt *testing.T) []byte {
 				input, err := PackSetFeeConfig(testFeeConfig)
 				require.NoError(tt, err)
 
 				return input
-			}(t),
+			},
 			SuppliedGas: SetFeeConfigGasCost,
 			ReadOnly:    true,
 			ExpectedErr: vmerrs.ErrWriteProtection.Error(),
 		},
 		"readOnly setFeeConfig with admin role fails": {
-			Caller: adminAddr,
-			Input: func(tt *testing.T) []byte {
+			Caller: allowlist.AdminAddr,
+			InputFn: func(tt *testing.T) []byte {
 				input, err := PackSetFeeConfig(testFeeConfig)
 				require.NoError(tt, err)
 
 				return input
-			}(t),
+			},
 			SuppliedGas: SetFeeConfigGasCost,
 			ReadOnly:    true,
 			ExpectedErr: vmerrs.ErrWriteProtection.Error(),
 		},
 		"insufficient gas setFeeConfig from admin": {
-			Caller: adminAddr,
-			Input: func(tt *testing.T) []byte {
+			Caller: allowlist.AdminAddr,
+			InputFn: func(tt *testing.T) []byte {
 				input, err := PackSetFeeConfig(testFeeConfig)
 				require.NoError(tt, err)
 
 				return input
-			}(t),
+			},
 			SuppliedGas: SetFeeConfigGasCost - 1,
 			ReadOnly:    false,
 			ExpectedErr: vmerrs.ErrOutOfGas.Error(),
 		},
-	} {
-		t.Run(name, func(t *testing.T) {
-			db := rawdb.NewMemoryDatabase()
-			state, err := state.New(common.Hash{}, state.NewDatabase(db), nil)
-			require.NoError(t, err)
-
-			// Set up the state so that each address has the expected permissions at the start.
-			SetFeeManagerStatus(state, adminAddr, allowlist.AdminRole)
-			SetFeeManagerStatus(state, enabledAddr, allowlist.EnabledRole)
-			require.Equal(t, allowlist.AdminRole, GetFeeManagerStatus(state, adminAddr))
-			require.Equal(t, allowlist.EnabledRole, GetFeeManagerStatus(state, enabledAddr))
-
-			if test.BeforeHook != nil {
-				test.BeforeHook(t, state)
-			}
-			blockContext := contract.NewMockBlockContext(testBlockNumber, 0)
-			accesibleState := contract.NewMockAccessibleState(state, blockContext, snow.DefaultContextTest())
-			if test.Config != nil {
-				Module.Configure(nil, test.Config, state, blockContext)
-			}
-			ret, remainingGas, err := FeeManagerPrecompile.Run(accesibleState, test.Caller, ContractAddress, test.Input, test.SuppliedGas, test.ReadOnly)
-			if len(test.ExpectedErr) != 0 {
-				require.ErrorContains(t, err, test.ExpectedErr)
-			} else {
-				require.NoError(t, err)
-			}
-
-			require.Equal(t, uint64(0), remainingGas)
-			require.Equal(t, test.ExpectedRes, ret)
-
-			if test.AfterHook != nil {
-				test.AfterHook(t, state)
-			}
-		})
 	}
+
+	allowlist.RunTestsWithAllowListSetup(t, Module, state.NewTestStateDB, tests)
+	allowlist.RunTestsWithAllowListSetup(t, Module, state.NewTestStateDB, allowlist.AllowListTests(Module))
```

Comment:
alternatively we can merge those two without requiring to run it with additional `allowlist.AllowListTests(Module)` tests. ` RunTestsWithAllowListSetup` can handle this merging.